public/components/demo.js

```diff
@@ -0,0 +1,19 @@
+var video1Complete = false;
+var video2Complete = false
+var video1 = document.getElementById('video1');
+var video2 = document.getElementById('video2');
+
+//Checks to see if the left-hand video has finished
+video1.addEventListener('ended', function () {
+    video1Complete = true;
+    //Checks to see if both videos are finished before reseting both videos
+    if (video2Complete == true) {
+        video1.play();
+        video2.play();
+    }
+  })
+//Checks to see if the right-hand video has finished
+video2.addEventListener('ended', function () {
+    //Signals that the video is completed
+    video2Complete = true;
+  })
```

Comment:
Your videos have "autoplay" set so the browser will attempt to download and play as soon as there is data. But your video1 event listener will wait until the first video is done playing before checking to see if video 2 is also done playing. 

This poses a problem: if video2 finishes AFTER video1, then when the "ended" event for video1 is emitted, video2complete will be false. By the time video2 finishes and video2complete is set to true, nothing will happen since video1's event listener was already called when when video1 is done.

I'd remove the autoplay attribute and use the [canplaythrough_event](https://developer.mozilla.org/en-US/docs/Web/API/HTMLMediaElement/canplaythrough_event) event instead. This waits until the entire video is loaded emits the event when the video "can play through." That way you won't ever get 2 videos playing out of sync. You will still need the event listeners for each videos, but you will need to make some modification such that both videos play only when both videos are done loading. Perhaps a helper function?

Reply:
The videos have the same length so there's not a huge wait time between both videos (it's about a split second). If you click the link at the top of the pr, the videos loop and the function works as intended. 

The problem is really just getting them to load at the same time. I'll look into that method you're talking about, and I'll try to implement it. Right now it works for 99% of cases, it's just the 1% of time they don't load together that's messing things up.

Reply:
@lillianchou94

Were you able to find something. I tried to use boolean to check both instances before playing in either event listener but it's not that great. Should I commit it or should we talk about it tomorrow?

Reply:
Hey there are still unresolved requests here. Could you please rebase and address the open comments.

About submitting: I think this code introduces a bug so maybe we should wait until we have some more clarity on the issue before submitting.